src/nvim/ops.c

```diff
@@ -5404,9 +5411,60 @@ static bool get_clipboard(int name, yankreg_T **target, bool quiet)
   return false;
 }
 
+
+/*
+ * Save and restore clip_unnamed before doing possibly many changes. This
+ * prevents accessing the clipboard very often which might slow down Vim
+ * considerably.
+*/
+
+/*
+ * Save clip_unnamed and reset it.
+ */
+void start_global_changes(void)
+{
+  if (++global_change_count > 1) {
+    return;
+  }
+  clip_unnamed_saved = clip_unnamed;
+  clipboard_needs_update = FALSE;
+
+  if (clip_did_set_selection) {
+    clip_unnamed = 0;
```

Comment:
`clip_unnamed` did become `cb_flags & CB_UNNAMEDMASK`  in the clipboard refactor. But rather than saving, hiding, and then restoring the value, can't `adjust_clipboard_name` (the only place where `cb_flags` have direct effect) just check the relevant condition directly ?